### PR TITLE
Document TOOMANYREQUESTS error code

### PR DESCRIPTION
**Pull request by aaronlehmann**
Add entries with this error code in registry/api/v2/descriptors.go.

### CODE REVIEW

**Comment by RichardScothern** [comment]
LGTM

**Comment by dmcgowan** [comment]
LGTM

File: docs/spec/api.md [hunk 1]
```diff
@@ -126,6 +126,13 @@ reference and shouldn't be used outside the specification other than to
 identify a set of modifications.
 
 <dl>
+  <dt>l</dt>
+  <dd>
+    <ul>
+      <li>Document TOOMANYREQUESTS error code.</li>
```
Comment (stevvooe) [inline docs/spec/api.md]: This should have ended up in autogenerated error table, rather than on each request.

**Comment by stevvooe** [comment]
@aaronlehmann This probably should have been put in the error code table, rather than on each request. Error response documentation is only for error specific to the request.

**Comment by aaronlehmann** [comment]
I patterned this after `unauthorized`, which is attached to the individusl endpoints. Do you think this error should be documented differently?

Can you point me to the error code table you mentioned?

**Comment by stevvooe** [comment]
The error code table is here https://github.com/docker/distribution/pull/1833/files#diff-b1b743dcaef6049308cb7cf5d689a3a1L1128.

I'm wondering if we need "generic" response examples to cover codes that can be encountered on any request. unauthorized falls into this category, as well.

**Comment by aaronlehmann** [comment]
Unfortunately the link doesn't seem to work for me. Could you please link to a source file instead of the diff?

**Comment by stevvooe** [comment]
https://github.com/docker/distribution/blob/master/docs/spec/api.md#errors-2

**Comment by SvenDowideit** [comment]
@aaronlehmann @dmcgowan what release will this be going out with?

do the docs changes apply to the currently release version?

**Comment by stevvooe** [comment]
@SvenDowideit As far as I understand, these change applies. The main issue here is verbosity but this can be pulled into the release.

**Comment by SvenDowideit** [comment]
@stevvooe @aaronlehmann @dmcgowan does this mean that this change will go out in v2.4.2 or in v2.5.0?

I'm presuming v2.5.0, and will give it that milestone, please change it if i'm wrong - we use the milestone to work out when to publish changes in the docs.

**Comment by stevvooe** [comment]
@SvenDowideit Doesn't matter. This error code has existed for some time, but was undocumented in the spec.

I'll defer to @RichardScothern if that's not the case.

**Comment by RichardScothern** [comment]
@SvenDowideit it's been present since v2.3.1

**Pull request merged**